confidence/models.py

```diff
@@ -338,5 +338,18 @@ def __radd__(self, other: typing.Sequence) -> typing.Sequence:
         return type(other)(list(other) + list(self))  # type: ignore
 
     def __repr__(self) -> str:
-        values = ', '.join(repr(value) for value in self)
-        return f'<{self.__class__.__module__}.{self.__class__.__name__} [{values}]>'
+        # use _source to avoid wrapping and resolving values
+        values = ', '.join(_repr_value(value) for value in self._source)
+        return f'{self.__class__.__module__}.{self.__class__.__name__}([{values}])'
+
+
+def _repr_value(value: typing.Any) -> str:
+    if isinstance(value, Mapping):
+        # represent a mapping as just its keys
+        keys = ', '.join(str(key) for key in value.keys())
+        return f'keys={{{keys}}}'
```

Comment:
Maybe we can draw some inspiration from how `dict` does it:
```
>>> {"Test": "test"}.keys()
dict_keys(['Test'])
```

So some more added square brackets (and perhaps even quotes).
And the parentheses instead of the curly brackets.

Of course this is just another option, and the nice thing is that a repr change will probably not trigger compatibility issues.

Reply:
I don't have any objections to the current approach, although the first time I saw it I associated the `{namespaces}` bit with a `set` literal.

Reply:
Using `([{keys}])`, the repr becomes

```plain
confidence.models.ConfigurationSequence([1, 2, mapping([namespaces])])
```

Definitely better, but clear enough?

Reply:
> ```
> confidence.models.ConfigurationSequence([1, 2, mapping([namespaces]), sequence([...])])
> ```
I like the solution like this. Do you think it is clear enough that the elements in the `mapping` are its keys?

And I don't think the explicit `sequence` is too verbose. Just `[...]` might be a bit too inconspicuous.

Reply:
But consider me pressing the "Approve" button even harder right now :-)

Reply:
I can only speak for myself, but I would like that better.